bip-musig2/reference.py

```diff
@@ -244,7 +244,11 @@ def nonce_agg(pubnonces: List[bytes]) -> bytes:
     for i in (1, 2):
         R_i_ = infinity
         for j in range(u):
-            R_i_ = point_add(R_i_, cpoint(pubnonces[j][(i-1)*33:i*33]))
+            try:
+                R_ij = cpoint(pubnonces[j][(i-1)*33:i*33])
+            except ValueError:
+                raise InvalidContributionError(j)
+            R_i_ = point_add(R_i_, R_ij)
```

Comment:
We could also just interpret invalid nonces as G? This would mean fewer error cases and fewer exceptions to handle for the caller. On the other hand, real applications will always need some DoS protection against too large or unsolicited messages, so not sure how much of a difference this is.

(Did we discuss this already? I vaguely remember.)

Reply:
I think it's better to know early if something is wrong and who's to blame. We have written up a proof why setting the aggnonce to G (which we do because we don't know who to blame at this point) and I suppose we'd need the same here/ We'd also keep this in mind forever to evaluate subtle interactions with protocol extensions in the future. I don't remember discussing this somewhere else.

Reply:
Yeah, I tend to agree. If the caller wants, they could still postpone error handling to the second round.